esp/src/src-react/components/forms/Fields.tsx

```diff
@@ -621,17 +629,19 @@ export const TargetFolderTextField: React.FunctionComponent<TargetFolderTextFiel
                 });
             }
         });
-    }, [machineDirectory, props.required]);
+    }, [dropzone, isContainer, machineDirectory, props.required]);
 
     React.useEffect(() => {
         const _fetchFolders = async () => {
             const folders = await fetchFolders(pathSepChar, machineAddress, machineDirectory, machineOS, 0);
             setFolders(folders);
         };
-        if (machineAddress && machineDirectory && machineOS) {
+        if (!isContainer && machineAddress && machineDirectory && machineOS) {
+            _fetchFolders();
+        } else if (machineDirectory && machineOS) {
```

Comment:
This logic looks wrong - the first "if" is just a special case of the "else if"?